### ISSUE TITLE
Text input oddities

### ISSUE THREAD

**Issue by 40110** [GTNewHorizons/GT-New-Horizons-Modpack#12766]
### Your Pack Version

2.3.0-RC1

### Your Server

SP

### Java Version

Java 19

### Type of Server

Single Player

### Your Expectation

Expected text input to behave normally.

Steps to reproduce:
- Run Arch Linux (haven't tested on other OSes)
- Set keyboard layout to DVORAK, example command for X: `setxkbmap -model abnt2 -layout br -variant dvorak` (for brazilian ABNT2 keyboard)
- Input text in-game: searching through NEI, writing in chat etc. (limiting max framerate makes effect more perceptible)

### The Reality

- Some characters are not input correctly, some are duplicated, others uppercased etc
- Whether on QWERTY or DVORAK, pressing CTRL seems to always input a comma character

### Your Proposal

Fix text input behavior to be the same as previous versions

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by mitchej123** [comment]
I've seen similar behavior locally recently

**Comment by Ethryan** [comment]
I have seen similar things happening if i try and press escape while writing in a textbox, I get the “” symbol written in then.

**Comment by r1tsuu** [comment]
I cannot reproduce this, maybe because i tried to set only DVORAK English and ABNT2 Brazilian (seems like there no way to set ABNT2 DVORAK Brazilian on macos, i tried to install this https://ports.macports.org/port/setxkbmap/ to run your command but it seems to not work).

May related to my PR https://github.com/GTNewHorizons/Hodgepodge/pull/188 but my mixin applies only on Mac, i don't think this can works wrong https://github.com/GTNewHorizons/Hodgepodge/pull/188/files#diff-9abfc174680df7590aae3e1630961abc404f534541209b49f7f106a844b7a9b0
Also tried set false (which you should have on Arch Linux) instead of system name check and mixin didn't load to the game, so there no problems in setApplyIf method. In config/Hodgepodge.cfg you can also try to set enableMacosCmdShortcuts to false, but i don't think it will do some.

**Comment by 40110** [comment]
Did you try it on the Java 9+ version? It's a lwjgl3ify issue

**Comment by mitchej123** [comment]
It's definitely a lwjgl3(ify) issue.  I'm seeing it on java8, and java17.  Seeing the issue with runClient with a direct checkout from lwjgly3ify.

I swear it was working just fine when I was running ubuntu, recently switched back to kde/kubuntu and it's flaring up.  Older versions don't seem to have the capitalization bug, but either dont' recognize text input or insert a different character.  Been trying to find a revision that works better to run  a git bisect on it, without much luck so far.

**Comment by r1tsuu** [comment]
> Did you try it on the Java 9+ version? It's a lwjgl3ify issue

Yes, i tried on java17 and i have no problems, i searched more and it only appears on some Linux DEs https://github.com/RecursiveG/Mc122477Fix

**Comment by mitchej123** [comment]
Running my launcher with the ENV var `XMODIFIERS="@im=null"` seems to have worked around it.

`XMODIFIERS="@im=null" prismlauncher` per https://bugs.mojang.com/browse/MC-122421#comment-608172

Still going to see if there's anything we can ship by default to bypass this
